### PR TITLE
Deploy non-main builds to a validation slot only, not staging->prod

### PR DESCRIPTION
**Pull request by directhex**
This failed in CI for an unrelated reason, the actual validation slot changes seem good

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -30,6 +30,11 @@ extends:
         timeoutInMinutes: 360
 
         variables:
+        - name: deploymentSlot
+          value: 'validation'
+        - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest'), eq(variables['Build.SourceBranch'], 'refs/heads/main')) }}:
```
Comment (ericstj) [inline azure-pipelines.yml:35]: IIUC staging is the production slot, and will only be used for internal non-PR builds of main?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -202,27 +207,28 @@ extends:
...
-            azureSubscription: SourceDotNet-Deployment-ARM
-            scriptLocation: inlineScript
-            scriptType: ps
-            inlineScript: >
-              deployment/cleanup-old-containers.ps1
-              -ResourceGroup source.dot.net
-              -WebappName netsourceindex
-              -StorageAccountName netsourceindex
\ No newline at end of file
+        - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest'), eq(variables['Build.SourceBranch'], 'refs/heads/main')) }}:
```
Comment (ericstj) [inline azure-pipelines.yml:210]: Consider adding a single variable for `publishToProduction` with the default value of `false` then use that to control both the slot name and these steps.

**Review by ericstj — COMMENTED** [review]
LGTM, just a minor suggestion to deduplicate the condition.

File: azure-pipelines.yml (hunk 1 above)
Comment (directhex) [reply azure-pipelines.yml:35]: The workflow is: builds from main deploy to `staging`, and are promoted to `prod` if they pass CI. in Theory, prod should never be broken by a bad build.

**Review by radical — APPROVED** [review]

**Pull request merged**